presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6965,6 +6982,17 @@ public void testExecute() throws Exception
     }
 
     @Test
+    public void testExecuteWithUsing()
+            throws Exception
+    {
+        String query = "SELECT a + ?, count(1) FROM (VALUES 1, 2, 3, 2) t(a) GROUP BY a + ? HAVING count(1) > ?";
```

Comment:
I think this is going to be problematic in the long term. I expect we're going to want to analyze queries at preparation time in the future by allowing clients to provide hints about the types of the arguments. Allowing this will make it impossible, since there's no way to tell that the `?` in the select expression and the group by expression are the same.

As I commented before, I think a more robust way of supporting this kind of usage is to allow for named parameters. Something like:

```sql
SELECT a + :foo, count(1) FROM (VALUES 1, 2, 3, 2) t(a) GROUP BY a + :foo HAVING count(1) > ?";
```